File: packages/sdk/sdk-ruby/tests/files_test.rb
require "minitest/autorun"
require_relative "../lib/index"

class FilesTest < Minitest::Test
  def test_upload_file_returns_payload
    client = AIStatsSdk::AIStats.new(
      api_key: "test",
      enable_deprecation_warnings: false
    )

    calls = []
    client.raw_client.define_singleton_method(:request) do |method:, path:, query: nil, headers: nil, body: nil|
      calls << [method, path, query, headers, body]
      { "id" => "file_123", "purpose" => "batch", "bytes" => 17 }
    end

    uploaded = client.upload_file(
      "purpose" => "batch",
      "file" => "data:application/json;base64,eyJ0ZXN0Ijp0cnVlfQ=="
    )

    assert_equal "file_123", uploaded["id"]
    assert_equal [["POST", "/files", nil, nil, {
      "purpose" => "batch",
      "file" => "data:application/json;base64,eyJ0ZXN0Ijp0cnVlfQ=="
    }]], calls
  end

  def test_retrieve_file_content_returns_bytes
    client = AIStatsSdk::AIStats.new(
      api_key: "test",
      enable_deprecation_warnings: false
    )

    calls = []
    client.raw_client.define_singleton_method(:request_bytes) do |method:, path:, query: nil, headers: nil, body: nil|
      calls << [method, path, query, headers, body]
      "{\"ok\":true}\n"
    end

    content = client.retrieve_file_content("file_123")

    assert_equal "{\"ok\":true}\n", content
    assert_equal [["get", "/files/file_123/content", nil, nil, nil]], calls
  end

  def test_retrieve_file_content_surfaces_request_errors
    client = AIStatsSdk::AIStats.new(
      api_key: "test",
      enable_deprecation_warnings: false
    )

    client.raw_client.define_singleton_method(:request_bytes) do |**_args|
      raise AiStats::Gen::RequestError.new(
        status_code: 404,
        status_message: "Not Found",
        response_body: "{\"error\":\"not found\"}"
      )
    end

    error = assert_raises(AiStats::Gen::RequestError) do
      client.retrieve_file_content("file_missing_123")
    end

    assert_equal 404, error.status_code
  end
end

File: packages/sdk/sdk-php/tests/files_test.php
<?php
declare(strict_types=1);

require_once __DIR__ . "/../src/index.php";

use Phaseo\Sdk\Phaseo;

function assert_true(bool $condition, string $message): void
{
    if (!$condition) {
        throw new RuntimeException($message);
    }
}

final class FakeFilesClient extends \Phaseo\Gen\Client
{
    public array $calls = [];
    public ?Throwable $error = null;

    public function __construct()
    {
        parent::__construct("https://example.test");
    }

    public function requestRaw(string $method, string $path, ?array $query = null, ?array $headers = null, $body = null): string
    {
        $this->calls[] = [$method, $path, $query, $headers, $body];
        if ($this->error) {
            throw $this->error;
        }
        return "{\"ok\":true}\n";
    }
}

$client = new Phaseo(
    apiKey: "test",
    basePath: "https://api.phaseo.app/v1",
    enableDeprecationWarnings: false
);

$fake = new FakeFilesClient();
$ref = new ReflectionClass($client);
$prop = $ref->getProperty("client");
$prop->setAccessible(true);
$prop->setValue($client, $fake);

$uploaded = $client->uploadFile([
    "purpose" => "batch",
    "file" => "data:application/json;base64,eyJ0ZXN0Ijp0cnVlfQ==",
]);
assert_true(($uploaded["ok"] ?? null) === true, "expected upload helper to return decoded payload");
assert_true($fake->calls[0] === [
    "POST",
    "/files",
    null,
    null,
    [
        "purpose" => "batch",
        "file" => "data:application/json;base64,eyJ0ZXN0Ijp0cnVlfQ==",
    ],
], "expected uploadFile to call POST /files");

$content = $client->retrieveFileContent("file_123");
assert_true($content === "{\"ok\":true}\n", "expected raw file content bytes");
assert_true($fake->calls[1] === ["GET", "/files/file_123/content", null, null, null], "expected requestRaw to be called for file content");

$fake->error = new \Phaseo\Gen\RequestException(404, "{\"error\":\"not found\"}");
try {
    $client->retrieveFileContent("file_missing_123");
    throw new RuntimeException("expected RequestException to be thrown");
} catch (\Phaseo\Gen\RequestException $error) {
    assert_true($error->getStatusCode() === 404, "expected RequestException status code");
}

echo "php files tests ok" . PHP_EOL;
